src/main/java/org/broadinstitute/barclay/argparser/TaggedArgument.java

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.barclay.argparser;
+
+import java.util.Map;
+
+/**
+ * Interface for arguments that can have tags and attributes. The command line argument parser
+ * looks for argument fields that implement this interface, and propagates tag names and
+ * attribute/value pairs to the field.
+ */
+public interface TaggedArgument {
+
+    /**
+     * Set the (optional) tag name for this instance.
+     * @param tagName
+     */
+    void setTag(String tagName);
+
+    /**
+     * Retrieve the tag name for this instance.
+     * @return String representing the tagName, or null if no name was set.
+     */
+    String getTag();
```

Comment:
I just noticed that the javadoc comments here are inconsistent with the rest of the implementation, which requires a tag name. The javadoc here should change.